tools/crm_resource_runtime.c

```diff
@@ -2063,10 +2068,33 @@ cli_resource_execute_from_params(pcmk__output_t *out, const char *rsc_name,
     return exit_code;
 }
 
+/*!
+ * \internal
+ * \brief Get the timeout the cluster would use for an action
+ *
+ * \param[in] rsc     Resource that action is for
+ * \param[in] action  Name of action
+ */
+static guint
+get_action_timeout(pcmk_resource_t *rsc, const char *action)
+{
+    long long timeout_ms = -1LL;
+    xmlNode *op = pcmk__find_action_config(rsc, action, 0, true);
+    GHashTable *meta = pcmk__unpack_action_meta(rsc, NULL, action, 0, op);
+
+    if ((pcmk__scan_ll(g_hash_table_lookup(meta, PCMK_META_TIMEOUT),
```

Comment:
I hadn't noticed that ... I think `unpack_timeout()` should definitely use `<= 0`. I'd expect it to get mapped to the default when it hits the services library, but I'm not sure it does. For sync actions (executing a file), `wait_for_sync_result()` would do a single `poll()` then immediately time out. I'm not sure async actions would time out. Systemd would be a little different (there's a 5s fudge factor added in some cases, and the DBus default will be used in other cases).

Seems like a "don't do that" situation

Reply:
It looks to me like a straightforward bug that just needs to be fixed in `unpack_timeout()`, I'll throw that in

Reply:
`controld_execd.c:construct_op()` allows timeout == 0 and then eventually defaults it to `op->interval_ms` if 0.

Same with `rules_alerts.c:get_meta_attrs_from_cib()`.

I don't think there's anywhere else outside of `crm_resource` that fetches and uses the `PCMK_META_TIMEOUT`. So I don't think `unpack_timeout` using 0 causes a problem for the live cluster, but it probably would for `crm_resource`.

However, if you change it to reject 0, that's going to cause `construct_op()` to use `PCMK_DEFAULT_ACTION_TIMEOUT_MS` as the default timeout instead of using `op->interval_ms` as the default timeout. Which might be fine.

Reply:
`unpack_timeout()` allowing 0 (if not a typo) might have been as a way to say "use the default"... but so would omitting the timeout altogether. Shrug

Reply:
> I don't think `unpack_timeout()` using 0 causes a problem for the live cluster

Ah it might for the action itself as you mentioned, rather than for the cluster's monitoring of the action

Reply:
I think it's helpful to be able to document a simple default, and the current behavior definitely seems like an oversight since it's handled differently in so many places without rhyme or reason